mixin/thanos/README.md

```diff
@@ -0,0 +1,154 @@
+
+To install:
+
+```shell
+go get github.com/google/go-jsonnet/cmd/jsonnet
+```
+
+### jsonnet-bundler
+
+Install this library in your own project with [jsonnet-bundler](https://github.com/jsonnet-bundler/jsonnet-bundler#install) (the jsonnet package manager):
+
+To install:
+
+```shell
+go get github.com/jsonnet-bundler/jsonnet-bundler/cmd/jb
+```
+
+> An e.g. of how to install a given version of this library: `jb install github.com/thanos-io/thanos/jsonnet/thanos-mixin@master`.
+
+## Use as a library
+
+To use the `thanos-mixin` as a dependency, simply use the jsonnet-bundler install functionality:
+```shell
+$ mkdir thanos-mixin; cd thanos-mixin
+$ jb init  # Creates the initial/empty `jsonnetfile.json`
+# Install the thanos-mixin dependency
+$ jb install github.com/thanos-io/thanos/jsonnet/thanos-mixin@master # Creates `vendor/` & `jsonnetfile.lock.json`, and fills in `jsonnetfile.json`
+```
+
+To update the `thanos-mixin` as a dependency, simply use the jsonnet-bundler update functionality:
+```shell
+$ jb update
+```
+
+#### Configure
+
+This project is intended to be used as a library. You can extend and customize dashboards and alerting rules by creating for own generators, such as the generators ([alerts.jsonnet](alerts.jsonnet) and [dashboards.jsonnet](dashboards.jsonnet)) that are use to create [examples](examples). Default parameters are collected in [defaults.jsonnet](defaults.jsonnet), feel free to modify and generate your own definitons.
+
+[embedmd]:# (defaults.libsonnet)
+```libsonnet
+{
+  query+:: {
+    jobPrefix: 'thanos-query',
+    selector: 'job=~"%s.*"' % self.jobPrefix,
+    title: '%(prefix)sQuery' % $.dashboard.prefix,
+  },
+  store+:: {
+    jobPrefix: 'thanos-store',
+    selector: 'job=~"%s.*"' % self.jobPrefix,
+    title: '%(prefix)sStore' % $.dashboard.prefix,
+  },
+  receive+:: {
+    jobPrefix: 'thanos-receive',
+    selector: 'job=~"%s.*"' % self.jobPrefix,
+    title: '%(prefix)sReceive' % $.dashboard.prefix,
+  },
+  rule+:: {
+    jobPrefix: 'thanos-rule',
+    selector: 'job=~"%s.*"' % self.jobPrefix,
+    title: '%(prefix)sRule' % $.dashboard.prefix,
+  },
+  compact+:: {
+    jobPrefix: 'thanos-compact',
+    selector: 'job=~"%s.*"' % self.jobPrefix,
+    title: '%(prefix)sCompact' % $.dashboard.prefix,
+  },
+  sidecar+:: {
+    jobPrefix: 'thanos-sidecar',
+    selector: 'job=~"%s.*"' % self.jobPrefix,
+    title: '%(prefix)sSidecar' % $.dashboard.prefix,
+  },
+  overview+:: {
+    title: '%(prefix)sOverview' % $.dashboard.prefix,
+  },
+  dashboard+:: {
+    prefix: 'Thanos / ',
+    tags: ['thanos-mixin'],
+    namespaceQuery: 'kube_pod_info',
+  },
+}
+```
+
+You can format your code using:
+```shell
+$ make jsonnet-format
+```
+
+## Examples
+
+This project is intended to be used as a library. However, it also provides drop-in examples to monitor Thanos.
+
+### Requirements
+
+#### gojsontoyaml
```

Comment:
No need for this, it's all handled by Makefile right?

Reply:
Yes, to be more explicit I'll document it and keep this anyway.